src/bun.js/node/node_fs.zig

```diff
@@ -3414,10 +3455,20 @@ pub const NodeFS = struct {
                             .buffer = Buffer.fromBytes(buf.items, bun.default_allocator, .Uint8Array),
                         },
                     },
-                    else => .{
-                        .result = .{
-                            .string = buf.items,
-                        },
+                    else => brk: {
+                        if (comptime string_type == .default) {
+                            break :brk .{
+                                .result = .{
+                                    .string = buf.items,
+                                },
+                            };
+                        } else {
+                            break :brk .{
+                                .result = .{
+                                    .null_terminated = buf.toOwnedSliceSentinel(0) catch unreachable,
```

Comment:
It’s fine for now but keep in mind that .toOwnedSlice potentially reallocates the array